### PR TITLE
AT-281 role test

### PR DESCRIPTION
**Pull request by JeffreyDevloo**
#281

### CODE REVIEW

File: ci/tests/general/general_disk.py [hunk 1]
```diff
@@ -172,6 +172,52 @@ def append_disk_role(partition, roles_to_add):
                                    roles=roles)
 
     @staticmethod
+    def set_disk_role(partition, roles):
```
Comment (kvanhijf) [inline ci/tests/general/general_disk.py]: 1. This method calls for another method and does nothing else only making things more complicated
2. Duplicate signature of this method

File: ci/tests/disklayout/disk_roles_test.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+class TestDiskRoles(object):
+    """
+    Disk roles testsuite
+    Testing situations covered by this class:
+     - Adding a role to a disk
+     - Removing all roles of a disk
+     - Appending a role to existing role on a disk
+     @TODO
+     - Remove a number of roles and leave x roles behind
```
Comment (pploegaert) [inline ci/tests/disklayout/disk_roles_test.py]: can we add this too?

File: ci/tests/disklayout/disk_roles_test.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        logger.info("Starting removal of disk roles")
+        for key, value in config.iteritems():
+            for disk_info in value['disks']:
+                roles_list = disk_info['roles']
+                disk_name = disk_info['disk_name']
+                logger.info("Fetching disk with diskname '{0}' for ip '{1}'".format(disk_name, key))
+                disk = GeneralStorageRouter().get_disk_by_ip(disk_name, key)
+                logger.info("Fetching or creating new partitions for disk '{0}'".format(disk.guid))
+                partition = GeneralDisk.partition_disk(disk)
+                if not number_of_roles_to_remain == 0:
```
Comment (pploegaert) [inline ci/tests/disklayout/disk_roles_test.py]: I'd prefer positive logic, which is more readable:
if ... == 0:
   do x
else:
   do y

File: ci/tests/disklayout/disk_roles_test.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+        This test will add a DB role to the sda disk of the storage router with the given IP
+        :param ip: IP address of a storage router. (Example:
+        :type ip: str
+
+        :param configuration: Dict that determines layout
+        :type configuration: dict
+        :return: None
+        """
+        # Start input validation
+        pattern = re.compile(r"^(?<!\S)((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\b|\.\b){7}(?!\S)$")
```
Comment (pploegaert) [inline ci/tests/disklayout/disk_roles_test.py]: IP validation could be useful for other tests too, so I'd suggest we create a general_network.py in ci/tests/general and add this as a static method there ...

File: ci/tests/general/general.py [hunk 5]
```diff
@@ -455,3 +455,7 @@ def validate_required_config_settings(settings=None):
 
         if len(missing_items) > 0:
             raise ValueError('Some required field are missing in autotest.cfg\n - {0}'.format('\n - '.join(missing_items)))
+
+    @staticmethod
+    def remove_list_from_list(all_values, values_to_remove):
+        return list(set(all_values) - set(values_to_remove))
```
Comment (pploegaert) [inline ci/tests/general/general.py]: add a newline - should be marked by pycharm too as missing - if not pls check pycharm's configuration

**Pull request merged**